Topics/Development_Process.md

```diff
@@ -21,3 +21,74 @@ The SOLID principles are:
  
  ## Resource that gives examples of the uses cases of SOLID principles
   LINK :  https://www.youtube.com/watch?v=_jDNAf3CzeY 
+
+
+## DevOps
+
+### Automated Testing
+
+Automated testing is an important part of industry software engineering development, and is
+virtually present at all major companies. While in certain companies, this role will be covered by a
+dedicated QA team, in others, it will be the responsibility of the developers themselves, e.g., at
+Amazon. Regardless, understanding and being able to write unit tests is important for a SWE to be
+able to communicate and standby their code.
+
+The automated testing process is part of the wider build/development process and is usually run as
+part of the CI/CD pipeline. The ultimate goal of automated testing is to ensure that the code
+functions as expected, and that any changes to the codebase do not break existing functionality.
+
+As background, most companies to implement their CI/CD pipeline with their Git implementation (of
+which are usually internal), and so we will be talking about unit testing and integration testing
+from the point of you pushing your code to a Git repository. So imagine that all of these tests will
+be executed from the moment that you push your code to your project repository.
+
+Further, there are usually "pipelines" for each project (e.g.,
+[AWS CodePipeline](https://docs.aws.amazon.com/codepipeline/latest/userguide/welcome-introducing.html)),
+and each of your code commits "flows" through your project pipeline. Generally, pipelines are
+usually composed of a series of "stages" that are executed sequentially. For example, after some
+cloud machine builds your code, the first testing stage might be to run the unit tests, and the
+second testing stage might be to run the integration tests. These setups vary from project to
+project, and you may be able to override passing criteria, but testing can lead credence to your
+implementation, and can support allow your code to promote (pass on to) production.
+
+![AWS CodePipeline Diagram](https://docs.aws.amazon.com/images/codepipeline/latest/userguide/images/pipeline-elements-workflow-application.png){fig-align="center"}
```

Comment:
`{fig-align="center"}` does not seem to be aligning on my end:

![image](https://user-images.githubusercontent.com/37385317/226796692-943922c3-15b4-4895-9157-9e065a9ef6e2.png)

Reply:
Removed